tests/dlq_test.go

```diff
@@ -112,6 +116,8 @@ func (s *dlqSuite) SetupSuite() {
 		dynamicconfig.HistoryTaskDLQEnabled: true,
 	}
 	s.dlqTasks = make(chan tasks.Task)
+	s.deleteBlockCh = make(chan interface{})
```

Comment:
How will you make this unblock in other tests? I think this is only implicitly working now because this channel is closed in `TestPurgeRealWorkflow` before `TestMergeRealWorkflow` runs, which seems kind of brittle to me. I think we'll need to do something a bit more clever to make this distinct between tests. For example, we can set  this instead in `SetupTest`, and close it at the beginning of every test except the CancelRunningMerge one.

Reply:
hmm right. I have updated the code.